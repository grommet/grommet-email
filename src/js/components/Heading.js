// (C) Copyright 2017 Hewlett Packard Enterprise Development LP

import React, { PropTypes } from 'react';
import { unit, halfUnit, fontWeight } from '../utils/Style';

const Heading = (props) => {
  const { tag: Tag, children } = props;
  const style = {
    fontWeight: fontWeight(Tag),
    margin: `${unit}px ${halfUnit}px`,
  };
  return (
    <Tag style={style}>
      {children}
    </Tag>
  );
};

Heading.propTypes = {
  children: PropTypes.any,
  tag: PropTypes.oneOf(['h1', 'h2', 'h3', 'h4', 'h5']).isRequired,
};

export default Heading;
