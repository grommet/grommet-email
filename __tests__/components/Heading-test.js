// (C) Copyright 2017 Hewlett Packard Enterprise Development LP

import React from 'react';
import renderer from 'react-test-renderer';

import Heading from '../../src/js/components/Heading';

describe('Heading', () => {
  it('has correct default options', () => {
    const component = renderer.create(
      <Heading tag="h1">Testing</Heading>,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
