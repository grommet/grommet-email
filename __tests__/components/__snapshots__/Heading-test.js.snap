// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Heading has correct default options 1`] = `
<h1
  style={
    Object {
      "fontWeight": 100,
      "margin": "24px 12px",
    }
  }
>
  Testing
</h1>
`;
